lib/puppet/provider/service/systemd.rb

```diff
@@ -30,6 +30,20 @@ def enabled?
     begin
       systemctl("is-enabled", @resource[:name])
     rescue Puppet::ExecutionFailure
+      # The masked state is equivalent to the disabled state in terms of
+      # comparison so we only care to check if it is masked if we want to keep
+      # it masked.
+      if @resource[:enable] == :mask
+        begin
+          output = systemctl('show', @resource[:name], '--property', 'LoadState', '--no-pager')
+          if output and (output.split('=').last[0,4] == 'mask')
+            return :mask
+          end
+        rescue Puppet::ExecutionFailure
+          # Don't worry about this failing, just return :false if it does.
```

Comment:
According to all the other providers, it's :false. This also held true with my testing, particularly with multi-state variables.